### PR TITLE
NixOS build instructions

### LINKED ISSUE #41: Clarify build instructions for NixOS
There are some problems when building Naggum for NixOS that should be researched (I've already built it but still cannot run).

### PR DESCRIPTION
**Pull request by ForNeVeR**
This PR adds a summary build information and more detailed instructions to build Naggum on NixOS. This will close #41.

### CODE REVIEW

**Comment by ForNeVeR** [comment]
@Minoru, could you please review these changes? You shouldn't care about `*.fsproj` / `*.sln` changes at all (our build servers will "review" that), but please valivate my language and instructions. You, as a more serious Linux user, could find them overly complicated or too hard to follow; please report that.

File: docs/Build.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+To build Naggum under Windows, just use Visual Studio or MSBuild like that:
+
+    $ cd naggum
+    $ nuget restore
+    $ msbuild /p:Configuration=Release Naggum.sln
+
+Linux
+-----
+
+See general build instructions for Linux in the file `.travis.yml` inside of the
```
Comment (Minoru) [inline docs/Build.md]: "of" is superfluous.

File: docs/Build.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+Naggum root source directory.
```
Comment (Minoru) [inline docs/Build.md]: "root source"? Too complicated; I suggest just "source".

File: docs/Build.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+You'll need [Mono][mono], [NuGet][nuget] and [F# Compiler][fsharp] installed in
+you current environment. Some of them may or may not be part of your Mono
```
Comment (Minoru) [inline docs/Build.md]: Why add "in your current environment"? You could stop at "installed".

File: docs/Build.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+installation; just make sure you have them all properly installed.
```
Comment (Minoru) [inline docs/Build.md]: I don't like so many repetitions of "install"; maybe rewrite the last part like this?

> just make sure you've got them all.

File: docs/Build.md (hunk 3 above)
Comment (ForNeVeR) [inline docs/Build.md]: My intention here was to clarify one noticeable thing. In NixOS you may install software to different environments/scopes (global, user-bound, shell-bound), and it'd be nice to say that we support any of these.

But now I agree with you, just "installed" should be enough; `nix-env` already installs software to the "current" environment (if the read cares about that at all).

File: docs/Build.md [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+Below is an example of setting up these tools on [NixOS Linux][nixos]; feel free
+to add instructions for any other non-standard distributives.
```
Comment (Minoru) [inline docs/Build.md]: "Distributions", not "destributives".

What's a "non-standard distribution"?

Comment (ForNeVeR) [inline docs/Build.md]: Yeah, silly typo.

Well, here I just trying to claim that the "standard" distribution is the one in that you can just install Mono and FSharp, and it'll work out of the box. Non-standard is the one that requires additional guide.

I think I'll replace this with "additional distributions".

Comment (Minoru) [inline docs/Build.md]: "Additional" would be superfluous, "add instructions for any other distributions" is fine.

Comment (ForNeVeR) [inline docs/Build.md]: Ok.

File: docs/Build.md [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+### NixOS Linux
+
+First of all, install `mono` and `fsharp` packages to your environment:
+
+    $ nix-env -i mono fsharp
+
+Please note that you need `fsharp` >= 4.0 to build Naggum, which is currently on
```
Comment (Minoru) [inline docs/Build.md]: Judging from the NixOS manual, you should say *in* the channel, not *on* it. Also, I'd say "unstable channel" is enough—it's obvious that we're talking about NixOS; and make it *the* channel, 'cause as far as I can tell, there's only one.

File: docs/Build.md [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+or build `fsharp` from [Nixpkgs][nixpkgs] source.
+
+According to the recommendations in
+[`patch-fsharp-targets`][patch-fsharp-targets] build helpers, you may want to
+set up the environment variable `FSharpTargetsPath` either globally or locally:
+
+    $ export FSharpTargetsPath=$(dirname $(which fsharpc))/../lib/mono/4.5/Microsoft.FSharp.Targets
+
+You'll need a NuGet executable. Unfortunately, there is no NixOS package for
+NuGet, so you need to download it manually:
```
Comment (Minoru) [inline docs/Build.md]: I'd use "have to" instead of "need to" to 1) avoid tautology; 2) make it clear that it's the requirement imposed on you, not your own desire :)

File: docs/Build.md [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+After that you can download the dependencies and build the project using
+`xbuild`:
+
+    $ mono nuget.exe restore Naggum.sln
+    $ xbuild /p:Configuration=Release /p:TargetFrameworkVersion="v4.5" Naggum.sln
+
+After that you should copy `FSharp.Core.dll` to the project output directory,
+because currently Nix have problems with concept of Mono global assembly cache:
+
+    $ FSHARP_CORE=~/.nix-profile/lib/mono/Reference\ Assemblies/Microsoft/FSharp/.NETFramework/v4.0/4.4.0.0/FSharp.Core.dll
```
Comment (Minoru) [inline docs/Build.md]: Are you sure that using tilde here is okay? As a rule, paths in environment variables are absolute. You can use `$HOME` instead, just to be on the safe side.

File: docs/Build.md [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+    $ echo Naggum.*/bin/Release | xargs -n 1 cp -f "$FSHARP_CORE"
```
Comment (Minoru) [inline docs/Build.md]: Not a review, but just a note of admiration: that's a cool trick! That's a very nice way to copy things into multiple directories.

File: docs/Build.md (hunk 8 above)
Comment (ForNeVeR) [inline docs/Build.md]: Yes, I am sure that it's okay, because it gets parsed by bash anyway in the next line. But thank you for suggestion, I'll replace it with `$HOME`.

File: docs/Build.md (hunk 9 above)
Comment (ForNeVeR) [inline docs/Build.md]: Found it [here](http://stackoverflow.com/questions/195655/how-to-copy-a-file-to-multiple-directories-using-the-gnu-cp-command) :)

File: docs/Build.md [hunk 10]
```diff
@@ -0,0 +1,72 @@
+Naggum Build Guide
+==================
+
+Windows
+-------
+
+To build Naggum under Windows, just use Visual Studio or MSBuild like that:
```
Comment (Minoru) [inline docs/Build.md]: I feel conflicted about "under" here. Google says that people say that a lot, but if you search for "on Windows", you get ten times more results. So that's just information; the decision is up to you.

File: docs/Build.md [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+because currently Nix have problems with concept of Mono global assembly cache:
+
+    $ FSHARP_CORE=~/.nix-profile/lib/mono/Reference\ Assemblies/Microsoft/FSharp/.NETFramework/v4.0/4.4.0.0/FSharp.Core.dll
+    $ echo Naggum.*/bin/Release | xargs -n 1 cp -f "$FSHARP_CORE"
+
+After that, you can run Naggum.Compiler, for example:
+
+    $ cd Naggum.Compiler/bin/Release/
+    $ mono Naggum.Compiler.exe ../../../tests/test.naggum
+    $ mono test.naggum
```
Comment (ForNeVeR) [inline docs/Build.md]: Oh, found an typo here: it should be `test.exe` instead of `test.naggum`.

File: docs/Build.md (hunk 8 above)
Comment (Minoru) [inline docs/Build.md]: Oh, right ._. Missed the wood for the trees.

**Comment by Minoru** [comment]
I'm through with reviewing and approve that for merge or whatever.

(Yay, I'm faster than AppVeyor!)
